addons/payment_asiapay/models/payment_acquirer.py

```diff
@@ -0,0 +1,47 @@
+from odoo import fields, models
+
+from .const import CURRENCY_MAPPING
+
+
+class PaymentAcquirer(models.Model):
+    _inherit = "payment.acquirer"
+
+    def _domain_asiapay_currency_id(self):
+        xmlids = ["base.{}".format(key) for key in CURRENCY_MAPPING]
+        return [('id', 'in', [self.env.ref(xmlid).id for xmlid in xmlids])]
+
+    provider = fields.Selection(selection_add=[("asiapay", "Asiapay")], ondelete={"asiapay": "set default"})
+    asiapay_merchant_id = fields.Char(
+        string="AsiaPay Merchant Partner ID",
+        required_if_provider="asiapay",
+        help="The Merchant Partner ID is used to connect to Asiapay merchant account.",
+    )
+    asiapay_currency_id = fields.Many2one(
+        "res.currency",
+        required_if_provider="asiapay",
+        string="AsiaPay Currency Code",
+        domain=_domain_asiapay_currency_id,
+    )
```

Comment:
If I understand correctly that a given AsiaPay account can only operate in a single currency, you should override the `_get_compatible_acquirers` method to filter out AsiaPay acquirers from the results when the payment currency does not match this one. See how it's done in other overrides ;)

Reply:
Originally we are trying to implement the multi currency API from Asiapay documentation. Thats why we are implementing in this way. However, turns out Asiapay multi currency does not work properly.

Reply:
As it's been quite a while since you reported the issue to them, wouldn't that make sense to start with simpler implementation and improve the module later in master, when they fix their API?